sdk/spring-sdk/src/main/scala/kalix/spring/boot/KalixConfiguration.scala

```diff
@@ -32,25 +34,32 @@ object KalixConfiguration {
     "In order to interact with a Kalix component, you should call it using the provided KalixClient."
 }
 
-@Configuration
+@AutoConfiguration(
+  // ReactiveWebServerFactoryAutoConfiguration is responsible for auto-configure the ReactiveWebServer
+  // it has different pre-defined configs, ie: Tomcat, Netty, Jetty. And it will load one of them if its server is in
+  // the classpath. It turns out that we need reactor-netty dependency for the WebClient, but we don't want Spring Boot
+  // to configure it as the main ReactiveWebServer. Therefore, we must require our AutoConfigure to be loaded
+  // before NettyReactiveWebServer is loaded
+  before = Array(classOf[ReactiveWebServerFactoryAutoConfiguration]))
```

Comment:
That was the biggest brain fkcu ever. 

The most surprising is that the tests was failing quite often in CI since I removed the `webflux-starter` (that brings in `reactor-netty`). 
Also surprising, when it failed, it was failing one the same tests. 

It turns out that the WebClient works (most of the times) without this dependency. And certainly when running locally.

We need reactor-netty in the production code because the WebClient is also used by the `KalixClient`, but if we leave `webflux-starter` in the classpath, Spring Boot will find two `ReactiveWebServers` configured and it will fail.

As soon as I brought `webflux-starter` back, all tests passed in CI.

Reply:
Ok, forgot to comment on important trick. 

If we load our configuration before `ReactiveWebServerFactoryAutoConfiguration`, the Netty one won't be loaded because there is a condition to prevent a second one to be loaded. But we must load ours first. 🤦